### DISCUSSION TITLE
Add new refinement level when restarting from restart file

### DISCUSSION THREAD

**Discussion by nscepi** [PrincetonUniversity/athena#465] [General]
Hello,
I am wondering if anyone has written a routine to add a level of static mesh refinement when restarting a simulation from a restart file? It seems like such on option could provide a considerable gain of computing time.
Thank you.

**Comment by c-white** [comment]
This is essentially AMR, but with a time- and space-dependent refinement criterion that triggers exactly once. What I don't know off the top of my head is whether AMR includes extra data in the restart file. If not, something like

```
<athena_recompiled_with_amr_condition> -r <restart_file> mesh/refinement=adaptive ...
```

might actually work. I think others have tried to do this, so maybe someone can chime in if this works.

**Reply by msbc** [reply]
Can't you change the SMR in the input file and restart?

**Reply by c-white** [reply]
I don't *think* that will auto-trigger prolongation. In fact, the code will really just read the restart file's embedded input file and construct the grid from that. I think that grid must have the same exact structure as the cell data in the file.

**Reply by tomidakn** [reply]
@c-white  is correct, changing the SMR grid structure in the input does not trigger refinement. I would suggest to use AMR as @c-white suggested.

**Comment by c-white** [comment]
Another possibility, certainly much more involved, is to manually create the appropriate restart file. The idea is to run a new simulation with the new SMR level for 0 steps, just to get the block structure in a new restart file. Then overwrite the data in that file with prolongated versions of the variables. This would require detailed knowledge of the restart format, the beginnings of which might be obtained with the help of this old script: https://www.physics.unlv.edu/astro/athena2019/scripts/restart_inspect.py

Hopefully it doesn't come to this.
